### ISSUE TITLE
how to control precision

### ISSUE THREAD

**Issue by erhuabushuo** [RedisJSON/RedisJSON#88]
127.0.0.1:6379> JSON.SET doc . '{"a": 0.1}'
OK
127.0.0.1:6379> JSON.GET doc .
"{\"a\":0.10000000000000001}"

I set the value 0.1,  after that I get the value 0.10000000000000001.

**Comment by itamarhaber** [comment]
Hello @erhuabushuo 

The current version of ReJSON does not provide the means to control precision. I agree this needs to be improved.

**Comment by mnunberg** [comment]
Usually it's better to store things like these as strings anyway, unless you actually need all that precision. We could control the precision at the module-level, but this means risking chopping off digits which the user *does* care about.

**Comment by itamarhaber** [comment]
@mnunberg indeed.

Also, IIRC, the JSON spec does not really say anything about numbers so stuff like precision and arbitrarily long numbers is pretty much a zoo out there.

That said, I suggest we keep this issue open as a reminder to ourselves.

**Comment by mnunberg** [comment]
Indeed, JSON is just an interchange format. It does not describe how to process data, but rather how to encode it. I think probably the simplest thing is to use %lf (the default representation) rather than use a specific precision. This should be good enough for most people.
